### PR TITLE
PLF-8383 : change news data model

### PR DESCRIPTION
**Pull request by thomasdelhomenie**
This PR moves News feature to integration project and apply the new data model based on CMS.

### CODE REVIEW

File: integ-news/service/src/main/java/org/exoplatform/news/webui/composer/UINewsActivityComposer.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import org.exoplatform.social.core.activity.model.ExoSocialActivityImpl;
+import org.exoplatform.social.core.application.PeopleService;
+import org.exoplatform.social.core.application.SpaceActivityPublisher;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.identity.provider.SpaceIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.space.model.Space;
+import org.exoplatform.social.webui.Utils;
+import org.exoplatform.social.webui.activity.news.UINewsActivity;
```
Comment (aelbenney) [inline integ-news/service/src/main/java/org/exoplatform/news/webui/composer/UINewsActivityComposer.java]: to be updated!

File: integ-news/service/src/main/java/org/exoplatform/news/webui/composer/UINewsActivityComposer.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      Space space = uiDisplaySpaceActivities.getSpace();
+
+      Identity spaceIdentity = Utils.getIdentityManager().getOrCreateIdentity(SpaceIdentityProvider.NAME,
+              space.getPrettyName(),
+              false);
+      activity = new ExoSocialActivityImpl(Utils.getViewerIdentity().getId(),
+              SpaceActivityPublisher.SPACE_APP_ID,
+              postedMessage,
+              null);
+      activity.setType(UINewsActivity.ACTIVITY_TYPE);
```
Comment (aelbenney) [inline integ-news/service/src/main/java/org/exoplatform/news/webui/composer/UINewsActivityComposer.java]: `ACTIVITY_TYPE` constant is deleted in `UINewsActivity`

File: integ-news/service/src/main/java/org/exoplatform/news/webui/composer/UINewsActivityComposer.java (hunk 1 above)
Comment (thomasdelhomenie) [reply integ-news/service/src/main/java/org/exoplatform/news/webui/composer/UINewsActivityComposer.java]: Good catch! Done.

File: integ-news/service/src/main/java/org/exoplatform/news/webui/composer/UINewsActivityComposer.java (hunk 2 above)
Comment (thomasdelhomenie) [reply integ-news/service/src/main/java/org/exoplatform/news/webui/composer/UINewsActivityComposer.java]: Done

File: integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+   */
+  public News getNews(String id) throws Exception {
+    SessionProvider sessionProvider = sessionProviderService.getSystemSessionProvider(null);
+
+    Session session = sessionProvider.getSession("collaboration", repositoryService.getCurrentRepository());
+
+    try {
+      Node node = session.getNodeByUUID(id);
+      return convertNodeToNews(node);
+    } catch (ItemNotFoundException e) {
```
Comment (azaoui) [inline integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: it's better to catch RepositoryException

Comment (azaoui) [inline integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: it's better to catch RepositoryException

File: integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    try {
+      News createdNews = newsService.createNews(news);
+
+      return Response.ok(createdNews).build();
+    } catch (Exception e) {
+      LOG.error("Error when creating the news " + news.getTitle(), e);
+      return Response.serverError().build();
+    }
+  }
+
```
Comment (azaoui) [inline integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: shall we add api operation for the added endpoint for swagger doc ?

File: integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    Node node = newsNode.addNode("illustration", "nt:file");
+    node.setProperty("exo:title", uploadedResource.getFileName());
+    Node resourceNode = node.addNode("jcr:content", "nt:resource");
+    resourceNode.setProperty("jcr:mimeType", uploadedResource.getMimeType());
+    resourceNode.setProperty("jcr:lastModified", Calendar.getInstance());
+    String fileDiskLocation = uploadedResource.getStoreLocation();
+    try(InputStream inputStream = new FileInputStream(fileDiskLocation)) {
+      resourceNode.setProperty("jcr:data", inputStream);
+      newsNode.save();
+    }
```
Comment (azaoui) [inline integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: we missed the removeUploadResource

File: integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java (hunk 3 above)
Comment (thomasdelhomenie) [reply integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: I catched only ItemNotFoundException to return null if the news does not exist, but I did not catch others to let the caller knows that something went wrong. WDYT ?

File: integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java (hunk 4 above)
Comment (thomasdelhomenie) [reply integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: Right!

File: integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java (hunk 5 above)
Comment (thomasdelhomenie) [reply integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: Good catch!

File: integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java (hunk 3 above)
Comment (azaoui) [reply integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: ok

File: integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java (hunk 4 above)
Comment (thomasdelhomenie) [reply integ-news/service/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: Done

File: integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java (hunk 5 above)
Comment (thomasdelhomenie) [reply integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: Done

**Comment by thomasdelhomenie** [comment]
Replaced by https://github.com/exodev/integration/pull/235

**Pull request closed without merging**